src/sfCommand.ts

```diff
@@ -138,10 +161,64 @@ export abstract class SfCommand<T> extends Command {
     };
   }
 
+  protected async catch(err: Error & { exitCode?: number }): Promise<Error & { exitCode?: number }> {
+    process.exitCode = process.exitCode ?? err.exitCode ?? 1;
+    this.log(this.formatError(err));
+    if (this.jsonEnabled()) {
+      CliUx.ux.styledJSON(this.toErrorJson(err));
+    }
+    return err;
+  }
+
+  /**
+   * Format errors and actions for human consumption. Adds 'Error:',
+   * When there are actions, we add 'Try this:' in blue
+   * followed by each action in red on its own line.
+   * If Error.code is present it is output last in parentheses
+   *
+   * @returns {string} Returns decorated messages.
+   */
+  protected formatError(error: Error & { actions?: string[]; code?: unknown }): string {
+    const colorizedArgs: string[] = [];
+    colorizedArgs.push(`${chalk.bold.red(messages.getMessage('error.error'))} ${error.message}`);
+    colorizedArgs.push(...this.formatActions(error));
+    if (error.stack && envVars.getString(SfCommand.SF_ENV) === Mode.DEVELOPMENT) {
+      colorizedArgs.push(chalk.red(`\n*** Internal Diagnostic ***\n\n${error.stack}\n******\n`));
+    }
+    if (error.code) {
+      colorizedArgs.push(chalk.bold(`\n(${error.code})`));
+    }
+
+    return colorizedArgs.join('\n');
+  }
+
+  /**
+   * Utility function to format actions lines
+   *
+   * @param error
+   * @param options
+   * @private
+   */
+  private formatActions(
+    error: (Error & { actions?: string[]; code?: unknown }) | string,
```

Comment:
Why not just add an `actions` and `code` property to the `Error` interface down below?